sklearnex/preview/ensemble/_forest.py

```diff
@@ -1163,78 +1083,6 @@ def _onedal_gpu_supported(self, method_name, *data):
 
         return patching_status
 
-    def _onedal_fit(self, X, y, sample_weight=None, queue=None):
-        if sp.issparse(y):
-            raise ValueError("sparse multilabel-indicator for y is not supported.")
-        if sklearn_check_version("1.2"):
-            self._validate_params()
-        else:
-            self._check_parameters()
-        if sample_weight is not None:
-            sample_weight = self.check_sample_weight(sample_weight, X)
-        if sklearn_check_version("1.0"):
-            self._check_feature_names(X, reset=True)
-        X = check_array(X, dtype=[np.float64, np.float32])
-        y = np.atleast_1d(np.asarray(y))
-        if y.ndim == 2 and y.shape[1] == 1:
-            warnings.warn(
-                "A column-vector y was passed when a 1d array was"
-                " expected. Please change the shape of y to "
-                "(n_samples,), for example using ravel().",
-                DataConversionWarning,
-                stacklevel=2,
-            )
-        y = check_array(y, ensure_2d=False, dtype=X.dtype)
-        check_consistent_length(X, y)
-        self.n_features_in_ = X.shape[1]
-        if not sklearn_check_version("1.0"):
-            self.n_features_ = self.n_features_in_
-
-        if self.oob_score:
-            err = "out_of_bag_error_r2|out_of_bag_error_prediction"
-        else:
-            err = "none"
-
-        onedal_params = {
-            "n_estimators": self.n_estimators,
-            "criterion": self.criterion,
-            "max_depth": self.max_depth,
-            "min_samples_split": self.min_samples_split,
-            "min_samples_leaf": self.min_samples_leaf,
-            "min_weight_fraction_leaf": self.min_weight_fraction_leaf,
-            "max_features": self.max_features,
-            "max_leaf_nodes": self.max_leaf_nodes,
-            "min_impurity_decrease": self.min_impurity_decrease,
-            "bootstrap": self.bootstrap,
-            "oob_score": self.oob_score,
-            "n_jobs": self.n_jobs,
-            "random_state": self.random_state,
-            "verbose": self.verbose,
-            "warm_start": self.warm_start,
-            "error_metric_mode": err,
-            "variable_importance_mode": "mdi",
-            "max_samples": self.max_samples,
-        }
-
-        # Lazy evaluation of estimators_
-        self._cached_estimators_ = None
-
-        self._onedal_estimator = self._onedal_regressor(**onedal_params)
-        self._onedal_estimator.fit(X, y, sample_weight, queue=queue)
-
-        self._save_attributes()
-
-        return self
-
-    def _onedal_predict(self, X, queue=None):
-        X = check_array(X, dtype=[np.float32, np.float64])
```

Comment:
This caused one of the test_common failures. For non-pandas/non-numpy data, it will cast to numpy array using the dtype list order as order of preference, and was casting things to float32. order is switched to [np.float64, np.float32] and centralized to BaseForest to minmize code duplication